agent/consul/leader.go

```diff
@@ -586,6 +586,53 @@ func (s *Server) initializeManagementToken(name, secretID string) error {
 	return nil
 }
 
+func (s *Server) upsertManagementToken(name, secretID string) error {
+	state := s.fsm.State()
+	if _, err := uuid.ParseUUID(secretID); err != nil {
+		s.logger.Warn("Configuring a non-UUID management token is deprecated")
+	}
+
+	_, token, err := state.ACLTokenGetBySecret(nil, secretID, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get %s: %v", name, err)
+	}
+	// Ignoring expiration times to avoid an insertion collision.
+	if token == nil {
```

Comment:
Might be cleaner to just early return here instead of wrapping this block of logic in an if. ex:

```
if token != nil {
  return nil
}
```